src/auth/permission_repository.zig

```diff
@@ -0,0 +1,244 @@
+const std = @import("std");
+const sqlite = @import("sqlite");
+const Database = @import("../database.zig").Database;
+const permissions = @import("permissions.zig");
+const Permission = permissions.Permission;
+const Role = permissions.Role;
+const UserRole = permissions.UserRole;
+
+pub const PermissionRepository = struct {
+    db: *Database,
+    allocator: std.mem.Allocator,
+
+    pub fn init(db: *Database, allocator: std.mem.Allocator) PermissionRepository {
+        return .{
+            .db = db,
+            .allocator = allocator,
+        };
+    }
+
+    /// Get all permissions for a user (via roles and direct grants)
+    pub fn getUserPermissions(self: *PermissionRepository, user_id: i64) !std.ArrayList(Permission) {
+        var perms: std.ArrayList(Permission) = .empty;
+        errdefer perms.deinit(self.allocator);
```

Comment:
Leaks if we return an error after adding some items. Need to iterate and clean them all up.